docs/static/persistent-queues.asciidoc

```diff
@@ -53,44 +51,46 @@ process:
-checkpoint file to track which events are successfully acknowledged (ACKed) as
-processed by Logstash. An event is recorded as ACKed in the checkpoint file if
-the event is successfully sent to the last output stage in the pipeline;
-Logstash does not wait for the output to acknowledge delivery. 
-
-During a normal, controlled shutdown (*CTRL+C*), Logstash finishes
+the persistent queue for processing. Logstash reads a batch of events from the
+persistent queue and sends them to the filter and output stages. As events
+are processed, Logstash records a checkpoint to track which events are
+successfully acknowledged (ACKed) as processed by Logstash. An event is
+recorded as ACKed if the event is successfully sent to the
+last output stage in the pipeline.
+
+During a normal, controlled shutdown (*CTRL+C* or SIGTERM), Logstash finishes
 processing the current in-flight events (that is, the events being processed by
 the filter and output stages, not the queued events), finalizes the ACKing
 of these events, and then terminates the Logstash process. Upon restart,
-Logstash uses the checkpoint file to pick up where it left off in the persistent
-queue and processes the events in the backlog. 
+Logstash uses the last checkpoint to resume processing the events in the backlog. 
 
 If Logstash crashes or experiences an uncontrolled shutdown, any in-flight
-events are left as unACKed in the persistent queue. Upon restart, Logstash will
-replay the events from its history, potentially leading to duplicate data being
-written to the output.
+events are left as unACKed in the persistent queue; during these abnormal
+terminations, Logstash is unable to safely save a new checkpoint, so the most
+recent checkpoint will be used on startup. Logstash will again resume from the
+last checkpoint the events from its history. This scenario can lead to some
+recent events being processed twice (once before the crash, and once upon
+restart and resuming).
 
 [[configuring-persistent-queues]]
 ==== Configuring Persistent Queues
 
 To configure persistent queues, you can specify the following options in the
 Logstash <<logstash-settings-file,settings file>>:
 
-* `queue.type`: Specify `persisted` to enable persistent queues. By default, persistent queues are disabled (`queue.type: memory`).
+* `queue.type`: Specify `persisted` to enable persistent queues. By default, persistent queues are disabled (default: `queue.type: memory`).
 * `path.queue`: The directory path where the data files will be stored. By default, the files are stored in `path.data/queue`. 
 * `queue.page_capacity`: The size of the page data file. The queue data consists of append-only data files separated into pages. The default size is 250mb. 
-* `queue.max_events`:  The maximum number of unread events that are allowed in the queue. The default is 0 (unlimited).
+* `queue.max_events`:  The maximum number of unACKed events that are allowed in the queue. The default is 0 (unlimited).
```

Comment:
sure, a queue event can be in the following state:
- unread (which also implies unacked)
- read & unacked (thus "in flight" in the worker) 
- read & acked
Now, read & acked events only exists because purging is done at page level. only when all events within a page are acked is it purged.